packages/es-components/src/components/containers/popover/Popover.js

```diff
@@ -147,15 +146,23 @@ function Popover(props) {
             escMsgRef.current.focus();
           }
         }, 100);
-        window.addEventListener('scroll', hidePopOnScroll);
       } else {
         triggerBtnRef.current.focus();
-        window.removeEventListener('scroll', hidePopOnScroll);
       }
     },
     [isOpen]
   );
 
+  useEffect(
+    () => {
+      if (isOpen) {
+        window.addEventListener('scroll', hidePopOnScroll);
+      }
+      return () => window.removeEventListener('scroll', hidePopOnScroll);
+    },
+    [isOpen]
```

Comment:
I'm a little mixed on having two `useEffect()` hooks on `isOpen`. Having the second makes it easier to read, admittedly, though it seems redundant

Reply:
It is a little weird at first, but I was going by the recommendation of the official docs, which states you can use multiple `useEffect` calls to separate concerns: https://reactjs.org/docs/hooks-effect.html#tip-use-multiple-effects-to-separate-concerns